web_custom_modifier/models/web_custom_modifier.py

```diff
@@ -0,0 +1,98 @@
+# © 2023 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, modules, tools
+
+
+class WebCustomModifier(models.Model):
+
+    _name = "web.custom.modifier"
+    _description = "Custom View Modifier"
+
+    model_ids = fields.Many2many(
+        "ir.model", "ir_model_custom_modifier", "modifier_id", "model_id", "Model"
+    )
+    type_ = fields.Selection(
+        [
+            ("field", "Field"),
+            ("xpath", "Xpath"),
+        ],
+        string="Type",
+        default="field",
+        required=True,
+    )
+    modifier = fields.Selection(
+        [
+            ("invisible", "Invisible"),
+            ("column_invisible", "Invisible (List Views)"),
+            ("readonly", "Readonly"),
+            ("force_save", "Force Save"),
+            ("required", "Required"),
+            ("selection_hide", "Hide Selection Item"),
+            ("widget", "Widget"),
+            ("limit", "Number of lines per page (List Views)"),
+            ("optional", "Optional"),
+        ],
+        required=True,
+    )
+    reference = fields.Char(required=True)
+    key = fields.Char()
+    active = fields.Boolean(default=True)
+    excluded_group_ids = fields.Many2many(
+        "res.groups",
+        "web_custom_modifier_excluded_group_rel",
+        "modifier_id",
+        "group_id",
+        "Excluded Groups",
+    )
+
+
+class WebCustomModifierWithCachedModifiers(models.Model):
+    """Add a cache for getting the modifiers.
+
+    The system cache is invalidated when any modifier record is added / modified / deleted.
+    """
+
+    _inherit = "web.custom.modifier"
+
+    @api.model
+    def create(self, vals):
+        new_record = super().create(vals)
+        modules.registry.Registry(self.env.cr.dbname).clear_caches()
+        return new_record
+
+    def write(self, vals):
+        super().write(vals)
+        modules.registry.Registry(self.env.cr.dbname).clear_caches()
```

Comment:
Why invalidating ALL models caches?
Which cache do you want to wipe? The `_get_cache` one? If yes, just clear its cache w/ `self._get_cache.clear_cache()`.